test/extended/apiserver/webhook.go

```diff
@@ -0,0 +1,386 @@
+			Ports: []v1.ContainerPort{{ContainerPort: containerPort}},
+		},
+	}
+	d := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   deploymentName,
+			Labels: podLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: podLabels,
+			},
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RollingUpdateDeploymentStrategyType,
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: podLabels,
+				},
+				Spec: v1.PodSpec{
+					TerminationGracePeriodSeconds: &zero,
+					Containers:                    containers,
+					Volumes:                       volumes,
+				},
+			},
+		},
+	}
+
+	deployment, err := client.AppsV1().Deployments(namespace).Create(context.Background(), d, metav1.CreateOptions{})
+	e2e.ExpectNoError(err, "creating deployment %s in namespace %s", deploymentName, namespace)
+	g.By("waiting for the deployment to be ready")
+	err = e2edeployment.WaitForDeploymentRevisionAndImage(client, namespace, deploymentName, "1", image)
+	e2e.ExpectNoError(err, "waiting for the deployment of image %s in %s in %s to complete", image, deploymentName, namespace)
+	err = e2edeployment.WaitForDeploymentComplete(client, deployment)
+	e2e.ExpectNoError(err, "waiting for the deployment status valid", image, deploymentName, namespace)
+
+	g.By("verifying the service has paired with the endpoint")
+	err = e2e.WaitForServiceEndpointsNum(client, namespace, serviceName, 1, 1*time.Second, 30*time.Second)
+	e2e.ExpectNoError(err, "waiting for service %s/%s have %d endpoint", namespace, serviceName, 1)
+}
+
+// createValidatingWebhook registers a ValidatingWebhookConfiguration that points to an always-deny webook for all resources matching objectSelector and namespaceSelector
+func createValidatingWebhook(client kubernetes.Interface, namespace string, namespaceSelector string, objectSelector string, configName string, serviceName string, servicePort int32) func() {
+	var err error
+	g.By("registering an always-deny validating webhook on all resources matching a special selector and namespace")
+
+	failurePolicy := admissionregistrationv1.Fail
+	sideEffectsNone := admissionregistrationv1.SideEffectClassNone
+
+	// this webhook denies all requests to Create that match objectSelector and namespaceSelector
+	config := &admissionregistrationv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        configName,
+			Annotations: map[string]string{"service.beta.openshift.io/inject-cabundle": "true"},
+		},
+		Webhooks: []admissionregistrationv1.ValidatingWebhook{
+			{
+				Name: "deny-all-webhook.openshift.io",
+				Rules: []admissionregistrationv1.RuleWithOperations{{
+					Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
+					Rule: admissionregistrationv1.Rule{
+						APIGroups:   []string{"*"},
+						APIVersions: []string{"*"},
+						Resources:   []string{"*"},
```

Comment:
okay, this is the same rule they are using (the customer case I am looking into). can we test this scenario?
- create a deploymentconfig
- and then `oc rollout latest dc/{name}`

or any equivalent test path that covers the ^ scenario.

Reply:
`deploymentconfig` is already covered by this test - https://github.com/openshift/origin/blob/master/test/extended/etcd/etcd_storage_path.go#L75

we do test all `*.openshift.io` resources